app_dart/lib/src/request_handlers/push_gold_status_to_github.dart

```diff
@@ -109,8 +109,12 @@ class PushGoldStatusToGithub extends ApiRequestHandler<Body> {
       }
 
       if (runsGoldenFileTests) {
-        if (cirrusCheckStatuses.any(kCirrusInProgressStates.contains)) {
-          // Checks are still running, we have to wait.
+        if (cirrusCheckStatuses.any(kCirrusInProgressStates.contains) ||
+            cirrusCheckStatuses.any(kCirrusFailedStates.contains) ||
```

Comment:
Shall we push gold status for finished runs, no matter the checkStatus is `Failed` or `succeeded`?

Reply:
Yes, there will always be a status reported to Github, either pending or completed.

I noticed this AM in the logs that we were checking gold status when there were failing tests on other shards, and that seemed like an unnecessary request in that case, so it should just report pending (without checking Gold) until all other failures are resolved.